engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -2424,4 +2435,103 @@ public int hashCode()
     return result;
   }
 
+  /**
+   * Return list of operators in topological sorted order. The Delay operators
+   * are considered as downstream operator.
+   * @return
+   */
+  @VisibleForTesting
+  protected List<OperatorMeta> getSortedOperators()
+  {
+    List<OperatorMeta> sortedOperators = new ArrayList<>(operators.size());
+    Stack<OperatorMeta> pendingNodes = new Stack<>();
+    Set<OperatorMeta> visited = new HashSet<>();
+
+    for (OperatorMeta n : this.getAllOperators()) {
+      pendingNodes.push(n);
+    }
+
+    while (!pendingNodes.isEmpty()) {
+      OperatorMeta n = pendingNodes.pop();
+
+      if (visited.contains(n)) {
+        continue;
+      }
+
+      boolean upstreamVisited = true;
+      for (Map.Entry<InputPortMeta, StreamMeta> entry : n.getInputStreams().entrySet()) {
```

Comment:
Instead of going with input streams, if you go with output streams it might be more efficient. See the algorithm marked depth-first search for topographical sorting here
https://en.wikipedia.org/wiki/Topological_sorting